spotify-artist-viewer/Networking/SARequestManager.m

```diff
@@ -2,22 +2,112 @@
 //  SARequestManager.m
 //  spotify-artist-viewer
 //
+//  Created by Sara Lieto on 6/9/15.
 //  Copyright (c) 2015 Intrepid. All rights reserved.
 //
 
 #import "SARequestManager.h"
+#import "SAArtist.h"
 
 @implementation SARequestManager
 
 + (instancetype)sharedManager {
-    // TODO: return a singleton instance of SARequestManager
-    return nil;
+    static dispatch_once_t onceToken;
+    static SARequestManager *sharedManager = nil;
+    dispatch_once(&onceToken, ^{
+        sharedManager = [SARequestManager new];
+    });
+    return sharedManager;
 }
 
-- (void)getArtistsWithQuery:(NSString *)query
-                    success:(void (^)(NSArray *artists))success
-                    failure:(void (^)(NSError *error))failure {
-    // TODO: make network calls to spotify API
+- (void) getArtistWithQuery:(NSString *) query success:(void (^)(NSArray *artists))success failure:(void(^)(NSError *error))failure{
+    //Construction of URL with current Search Bar Text (query)
+    NSString *urltest = @"https://api.spotify.com/v1/search?q=";
+    NSString *url = [urltest stringByAppendingString:query];
+    NSString *type = @"&type=artist";
+    NSString *urlWithType = [url stringByAppendingString:type];
+    NSURL *requestURL = [NSURL URLWithString:urlWithType];
+    NSMutableURLRequest *urlRequest = [[NSMutableURLRequest alloc] initWithURL:requestURL];
+    [urlRequest setHTTPMethod:@"GET"];
+    
+    NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration];
+    NSMutableArray *artists = [[NSMutableArray alloc] init];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:urlRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        NSDictionary *result = [NSJSONSerialization JSONObjectWithData:data
+                                                               options:0
+                                                                 error:NULL];
+
+        if([result isKindOfClass:[NSDictionary class]]){
+            NSDictionary *itemDictionary = [result objectForKey:@"artists"];
+            for(NSDictionary *item in [itemDictionary valueForKey:@"items"]){
+                SAArtist *artist = [[SAArtist alloc]init];
+                
+                NSDictionary *imgArray = [item objectForKey:@"images"];
+                artist.artistName = [item valueForKey:@"name"];
+                artist.artistUri = [item valueForKey:@"uri"];
+                NSArray *tempImgUrl = [imgArray valueForKey:@"url"];
+                if(tempImgUrl.count != 0){
+                    artist.imgURL = [tempImgUrl objectAtIndex:0];
+                }
+                [artists addObject:artist];
+
```

Comment:
Try to not use the data type in a variable name like itemDictionary or imgArray. I this section you used the name imgArray to hold and array, but you defined the data type as NSDictionary.